### DISCUSSION TITLE
optionally setting aws default profile as part of the workflow.

### DISCUSSION THREAD

**Discussion by nickforsythbarr** [fwdcloudsec/granted#686] [Feature Requests, Ideas]
I avoid profiles because I use SSO, and I don't like having to specify `--profile` with each command.

When assuming a profile with granted, we still have to do this which is sad. I understand the reasons why you'd want to avoid setting defaults with persistent credentials as to avoid mistakes etc. (e.g which  profile am I in?)

Can you please consider having one more prompt along the lines of "Would you like to set this profile as default?" and if Yes |

then run the platform equivalent of: `export AWS_DEFAULT_PROFILE="$CHOSENPROFILE"`

**Comment by chrnorm** [comment]
@nickforsythbarr what’s your usual workflow with Granted — what commands do you run in a typical AWS CLI workflow? And how do you like to structure your config file?

Personally I use `assume <profile name>` which exports `AWS_PROFILE` for me, which means I don’t need to use `—profile`.

Here’s what my profiles look like, for reference:

```
[profile web-application-prod/ViewOnlyAccess]
granted_sso_start_url      = https://d-12345abcdef.awsapps.com/start
granted_sso_region         = ap-southeast-2
granted_sso_account_id     = 123456789012
granted_sso_role_name      = ViewOnlyAccess
common_fate_generated_from = common-fate/v1
credential_process         = granted credential-process --profile web-application-prod/ViewOnlyAccess
```

**Comment by nickforsythbarr** [comment]
theres is a high chance I'm just not using it correctly. The setup is a bit confusing, and today is my first day.

Typically (previous to using this) I've just been pasting temp creds into terminals and then away I go, I've never bothered with the aws sso config before.

So Im not entirely sure how its expected to behave. I have read the guide through a couple of times and can't actually see what I'm missing.

I'm going to ask a coworker for help later on today and we'll see how we go. will close once sorted.

**Reply by chrnorm** [reply]
No problems. Feel free to share a version of your `~/.aws/config` file in this thread (you can redact any sensitive identifiers like account IDs) and I’d be happy to help you get started.

It seems like our docs could be made clearer here too. We’ve actually recently open sourced them (https://github.com/common-fate/docs), so any specific feedback you have on them would be very welcome.

Also, something you could try is using [granted sso populate](https://docs.commonfate.io/granted/usage/automatic-config-generation) which will generate AWS profiles for you.

**Reply by chrnorm** [reply]
PS - congrats on the new role! 😀

**Reply by nickforsythbarr** [reply]
sorry I meant first day using this solution :)

**Reply by nickforsythbarr** [reply]
Im trying to get onto slack, and then maybe someone can have a quick look over screenshare if appropriate

**Reply by nickforsythbarr** [reply]
After running through this with one of your colleagues it looks like it could actually be a bug.

https://github.com/common-fate/granted/issues/687

Some more staff my end have duplicate setups so can confer with them and confirm if its definitely a pattern.

**Reply by nickforsythbarr** [reply]
Can confirm on another ubuntu 24.04 same issue, no entry added in profile (assumego hadnt been activated), and despite manual entry in this file it didnt work oddly. had to add it to users' bashrc.
